test/tests/functional/pbs_provisioning.py

```diff
@@ -158,30 +154,23 @@ def test_execjob_begin_hook_on_os_provisioned_job(self):
                                 max_attempts=20,
                                 interval=1)
         self.assertTrue(rv)
-        self.server.delete(jid, extend='force')
 
     def test_app_provisioning(self):
         """
-        Test the application provisioning"
+        Test application provisioning
         """
         a = {'resources_available.aoe': 'App1'}
-        rv = self.server.manager(
+        self.server.manager(
             MGR_CMD_SET, NODE, a, id=self.hostA, expect=True)
-        self.assertTrue(rv)
 
         job = Job(TEST_USER1, attrs={ATTR_l: 'aoe=App1'})
         job.set_sleep_time(1)
         jid = self.server.submit(job)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
-        rv = self.server.log_match(
-            "fake calling Application provisioning script",
+        self.server.log_match(
+            "fake calling application provisioning script",
             max_attempts=20,
             interval=1)
-        self.assertTrue(rv)
-        self.server.delete(jid, extend='force')
 
     def tearDown(self):
-        rc = self.server.manager(MGR_CMD_DELETE, HOOK, None, self.hook_list[0])
-        self.assertEqual(rc, 0)
-        rc = self.server.manager(MGR_CMD_DELETE, HOOK, None, self.hook_list[1])
-        self.assertEqual(rc, 0)
+        self.server.revert_to_defaults()
```

Comment:
No need to call this explicitly in tearDown().  It is called in setUp().

Reply:
This is just clean up on exit.

Reply:
Calling it in tearDown() will make it be called twice between tests.  This is wasteful and causes the tests to run slower.

Reply:
But when it is removed from tearDown, previous unfinished job is causing error in the setUp of next testcase.
`c7-dev: /opt/pbs/bin/qmgr -c delete node @default`
`err: ['qmgr obj= svr=default: Cannot delete busy object', 'qmgr: Error (15027) returned from server']`

Reply:
That doesn't make any sense.  Pretty much the first thing setUp() does is call revert_to_defaults() on the server object.  Why would calling it twice help?  Are there nodes provisioning and the added revert_to_defaults() causes a delay which allows it to finish?

Reply:
Oh! my overriding setUp() is not calling parent's setUp(), so there will be no multiple calls to revert_to_defaults().

Reply:
It's dangerous to not call the parent's setUp().  PTL will assume that the parent's setUp() will be called.  Any number of things could be added to setUp() in the future with that assumption in mind.  Your test can start failing because of it.  Why exactly are you skipping the parent's setUp()?

Reply:
I thought it is not required since it reverts all the services, but calling parent's setup will help in future. I have updated the code.